### DISCUSSION TITLE
Map grafana authentication, to a datasource authentication

### DISCUSSION THREAD

**Discussion by j-dc** [grafana/grafana#38335] [Data Sources]
**What would you like to be added**:
The possibility to map grafana users onto datasource users.

**Why is this needed**:
1) To prevent users from seeing the data of other users (GDPR). 
2) To make this manageable. 

I have a datasource that is capable of a per-item security.   (SQL-Server).  Based on the user executing a query you can get a different result.  To use this feature in Grafana,  I must create a datasource per User, and copy all the dashboards per user..  etc..
In my use case there are 30 users..  that's barely manageable (without human mistakes),   but in the future we will have +100 users..    It would be magnificent if I could map a grafana user to a datasource user.

**Comment by ying-jeanne** [comment]
Hi @j-dc, thanks for reporting the issue. Unfortunately, we might adding feature as so in the future but not anytime soon. What you could try instead:
1. use a combination of an auth proxy infront of grafana paired with a proxy infront of the datasource.
2. using oauth and browser based access.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
